immuni_analytics/celery/authorization/tasks/authorize_analytics_token.py

```diff
@@ -0,0 +1,163 @@
+#   Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#   Please refer to the AUTHORS file for more information.
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as
+#   published by the Free Software Foundation, either version 3 of the
+#   License, or (at your option) any later version.
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#   GNU Affero General Public License for more details.
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import random
+
+from immuni_analytics.celery.authorization.app import celery_app
+from immuni_analytics.core import config
+from immuni_analytics.core.managers import managers
+from immuni_analytics.helpers.device_check import fetch_device_check_bits, set_device_check_bits
+from immuni_analytics.helpers.redis import (
+    get_authorized_tokens_redis_key_current_month,
+    get_authorized_tokens_redis_key_next_month,
+)
+from immuni_common.core.exceptions import ImmuniException
+from immuni_common.models.enums import Environment
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DiscardAnalyticsTokenException(ImmuniException):
+    """Raised when the device cannot authorize an analytics token."""
+
+
+@celery_app.task()
+def authorize_analytics_token(analytics_token: str, device_token: str) -> None:  # pragma: no cover
+    """
+     Celery doesn't support async functions, so we wrap it around asyncio.run.
+     """
+    asyncio.run(_authorize_analytics_token(analytics_token, device_token))
+
+
+async def _authorize_analytics_token(analytics_token: str, device_token: str) -> None:
+    try:
+        # NOTE: bandit complains about using random.uniform stating "[B311:blacklist] Standard
+        #  pseudo-random generators are not suitable for security/cryptographic purposes.".
+        #  The "waiting for a random time" is not a security/cryptographic action, thus the issue is
+        #  intentionally waived.
```

Comment:
Can't we use the recommended random packages anyway? 😄

Reply:
The `secrets` lib comes with a `randbelow(n)` method, which will give you a random number between `[0, n)`. Unfortunately, we ain't interested in `0`.

One alternative would be using either `random` or `secrets`'s `choice(sequence)` method, specifically:
```python
random.choice(c for c in range(1, config.CHECK_TIME_SECONDS + 1))
```

Another alternative would be keeping it as-is since, as explained in the note, it shouldn't be a security concern after all.

Reply:
There are also methods from the random library that bandit won't complain about, like normalvariate or similar.

Reply:
randbelow(n) + 1?

Reply:
> There are also methods from the random library that bandit won't complain about, like normalvariate or similar.

We tried those. 🙏 Yet, here you want a uniform distribution.

Reply:
let's keep `random.uniform` then 👍

Reply:
There's actually `randint`, trying it.

Reply:
✅